app/models/user.rb

```diff
@@ -12,4 +12,5 @@ class User < ApplicationRecord
   validates :phone, format: { with: VALID_TELEPHONE_REGEX}
   # Has offspring
   has_many :offsprings, dependent: :destroy
+  has_many :assignments
```

Comment:
It is quite likely that you don't need this. You don't need to go from user to assignments, only from assignment to user